platform/ui/src/components/ContextMenu/ContextMenuController.tsx

```diff
@@ -0,0 +1,235 @@
+import * as ContextMenuItemsBuilder from './ContextMenuItemsBuilder';
+import ContextMenu from './ContextMenu';
+
+type Obj = Record<string, unknown>;
+type Point = {
+  x: number;
+  y: number;
+};
+
+/**
+ * The context menu controller is a helper class that knows how
+ * to manage context menus based on the UI Customization Service.
+ * There are a few parts to this:
+ *    1. Basic controls to manage displaying and hiding context menus
+ *    2. Menu selection services, which use the UI customization service
+ *       to choose which menu to display
+ *    3. Menu item adapter services to convert menu items into displayable and actionable items.
+ */
+export default class ContextMenuController {
+  commandsManager: Obj;
+  services: Obj;
+  menuItems: [];
+
+  constructor(servicesManager: Obj, commandsManager: Obj) {
+    this.services = servicesManager.services as Obj;
+    this.commandsManager = commandsManager;
+  }
+
+  closeViewerContextMenu() {
+    this.services.uiDialogService.dismiss({ id: 'context-menu' });
+  }
+
+  showContextMenu(
+    contextMenuProps: Obj,
+    activeViewerElement: Obj,
+    defaultPointsPosition
+  ): void {
+    if (!this.services.uiDialogService) {
+      console.warn('Unable to show dialog; no UI Dialog Service available.');
+      return;
+    }
+
+    const {
+      event,
+      subMenu,
+      menuId,
+      menus,
+      refs,
+      checkProps,
+    } = contextMenuProps;
+
+    const items = ContextMenuItemsBuilder.getMenuItems(
+      checkProps || contextMenuProps,
+      event,
+      menus,
+      refs,
+      menuId
+    );
+
+    this.services.uiDialogService.dismiss({ id: 'context-menu' });
+    this.services.uiDialogService.create({
+      id: 'context-menu',
+      isDraggable: false,
+      preservePosition: false,
+      preventCutOf: true,
+      defaultPosition: ContextMenuController._getDefaultPosition(
+        defaultPointsPosition,
+        event?.detail,
+        activeViewerElement
+      ),
+      event,
+      content: ContextMenu,
+
+      onClickOutside: () =>
+        this.services.uiDialogService.dismiss({ id: 'context-menu' }),
+
+      contentProps: {
+        items,
+        checkProps,
+        menus,
+        event,
+        subMenu,
+        eventData: event?.detail,
+        refs,
+        onRunCommands: item => {
```

Comment:
so `onRunCommands` runs each command? maybe we name it `run`?

Reply:
or `onRun`

Reply:
The actionType is RunCommands, so the name for the function has to be onRunCommands.
I can change to onRun and make the actionType 'Run' if you want?  I think RunCommands is more descriptive here, but up to you.